rapids-4-spark-tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/Analysis.scala

```diff
@@ -0,0 +1,142 @@
+
+
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.tool.profiling
+
+import scala.collection.mutable.ArrayBuffer
+
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.rapids.tool.profiling._
+
+
+/**
+ * Analysis does analysis on the DataFrames
+ * from object of ApplicationInfo
+ */
+class Analysis(apps: ArrayBuffer[ApplicationInfo]) {
+
+  require(apps.nonEmpty)
+  private val fileWriter = apps.head.fileWriter
+
+  // Job Level TaskMetrics Aggregation
+  def jobMetricsAggregation(): Unit = {
+    if (apps.size == 1) {
+      fileWriter.write("Job level aggregated task metrics:")
+      apps.head.runQuery(apps.head.jobMetricsAggregationSQL + " order by Duration desc")
+    }
+    else {
+      var query = ""
+      var i = 1
+      for (app <- apps) {
+        if (i == 1) {
```

Comment:
We can remove the `i` variable and change this conditional to `if (query.isEmpty)` ?

Reply:
Done